x-pack/plugins/lens/public/heatmap_visualization/visualization.tsx

```diff
@@ -0,0 +1,394 @@
+                        : [],
+                      cellHeight: state.gridConfig.cellHeight ? [state.gridConfig.cellHeight] : [],
+                      cellWidth: state.gridConfig.cellWidth ? [state.gridConfig.cellWidth] : [],
+                      // cells
+                      isCellLabelVisible: [state.gridConfig.isCellLabelVisible],
+                      // Y-axis
+                      isYAxisLabelVisible: [state.gridConfig.isYAxisLabelVisible],
+                      yAxisLabelWidth: state.gridConfig.yAxisLabelWidth
+                        ? [state.gridConfig.yAxisLabelWidth]
+                        : [],
+                      yAxisLabelColor: state.gridConfig.yAxisLabelColor
+                        ? [state.gridConfig.yAxisLabelColor]
+                        : [],
+                      // X-axis
+                      isXAxisLabelVisible: [state.gridConfig.isXAxisLabelVisible],
+                    },
+                  },
+                ],
+              },
+            ],
+          },
+        },
+      ],
+    };
+  },
+
+  toPreviewExpression(state, datasourceLayers): Ast | null {
+    const datasource = datasourceLayers[state.layerId];
+
+    const originalOrder = datasource.getTableSpec().map(({ columnId }) => columnId);
+    // When we add a column it could be empty, and therefore have no order
+
+    if (!originalOrder) {
+      return null;
+    }
+
+    return {
+      type: 'expression',
+      chain: [
+        {
+          type: 'function',
+          function: FUNCTION_NAME,
+          arguments: {
+            title: [''],
+            description: [''],
+            xAccessor: [state.xAccessor ?? ''],
+            yAccessor: [state.yAccessor ?? ''],
+            valueAccessor: [state.valueAccessor ?? ''],
+            legend: [
+              {
+                type: 'expression',
+                chain: [
+                  {
+                    type: 'function',
+                    function: LEGEND_FUNCTION,
+                    arguments: {
+                      isVisible: [false],
+                      position: [],
+                    },
+                  },
+                ],
+              },
+            ],
+            gridConfig: [
+              {
+                type: 'expression',
+                chain: [
+                  {
+                    type: 'function',
+                    function: HEATMAP_GRID_FUNCTION,
+                    arguments: {
+                      // grid
+                      strokeWidth: [0],
+                      // cells
+                      isCellLabelVisible: [false],
+                      // Y-axis
+                      isYAxisLabelVisible: [false],
+                      // X-axis
+                      isXAxisLabelVisible: [false],
+                    },
+                  },
+                ],
+              },
+            ],
+          },
+        },
+      ],
+    };
+  },
+
+  getErrorMessages(state) {
```

Comment:
The error messages here start as soon as you open the heatmap configuration. There should be a special case for a completely empty chart so that it's only an error if there's partial config, but a fully empty config is fine.

Reply:
Fixed in https://github.com/elastic/kibana/pull/97978/commits/c31d3367e558e73e71d0d327e333d5a6725a98ca